src/src/PluginStructs/P036_data_struct.cpp

```diff
@@ -760,19 +769,22 @@ uint8_t P036_data_struct::display_scroll(ePageScrollSpeed lscrollspeed, int lTas
         ScrollingPages.LineIn[j] = ScrollingPages.LineIn[j].substring(iCharToRemove);
       }
       # ifdef PLUGIN_036_DEBUG
+      String log;
 
-      if (loglevelActiveFor(LOG_LEVEL_INFO)) {
-        log  = F("Line: "); log += (j + 1);
+      if (loglevelActiveFor(LOG_LEVEL_INFO) &&
+          log.reserve(128)) {
+        log += F("Line: "); log += (j + 1);
         log += F(" LineIn: "); log += LineInStr;
         log += F(" Length: "); log += strlen;
         log += F(" PixLength: "); log += PixLengthLineIn;
         log += F(" AvgPixPerChar: "); log += fAvgPixPerChar;
         log += F(" CharsRemoved: "); log += iCharToRemove;
         addLog(LOG_LEVEL_INFO, log);
-        log  = F(" -> Changed to: "); log += ScrollingPages.LineIn[j];
+        log.clear();
+        log += F(" -> Changed to: "); log += ScrollingPages.LineIn[j];
```

Comment:
Just FYI.
When assigning a flash string, then there is no std::move operator used.
When assigning the result from a function returning a String object, the assignment is using `operator=(String&&)` thus performing a std::move and thus undoing the reserve.